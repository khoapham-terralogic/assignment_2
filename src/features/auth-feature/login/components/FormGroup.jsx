import React from 'react';
import PropTypes from 'prop-types'

const FormGroup = ({ type, label, labelName, handleShow, isShow, className, value, placeholder, onChange, frontSvg, rearSvg }) => {
    return (
        <div className="form-group">
            <label htmlFor={label}>{labelName}</label>
            <div className={`${label}-group`}>
                <img className="form-group-frontSvg" src={frontSvg} alt="" />
                <input type={type}
                    className={`form-control ${label}-group`}
                    placeholder={placeholder}
                    value={value}
                    onChange={onChange}
                />
                <img className="form-group-rearSvg" onClick={handleShow} src={rearSvg} alt="" />
            </div>
        </div>
    );
}

FormGroup.propTypes = {
    type: PropTypes.string.isRequired,
    className: PropTypes.string,
    value: PropTypes.any,
    id: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    frontSvg: PropTypes.any,
    rearSvg: PropTypes.any,
    label: PropTypes.string.isRequired,
    labelName: PropTypes.string.isRequired,
    handleShow: PropTypes.func
}

FormGroup.defaultProps = {
    type: "text",
}

export default FormGroup;
