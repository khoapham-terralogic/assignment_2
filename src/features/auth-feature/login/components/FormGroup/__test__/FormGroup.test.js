import React from 'react'
import FormGroup from '..'
import { shallow, mount } from 'enzyme'

const props = {
    type: "text",
    label: "email",
    labelName: "email",
    frontSvg: "../Images/frontSvg.svg",
    rearSvg: "../Images/rearSvg.svg",
    rearSvgShow: "../Images/rearSvgShow.svg",
}

describe("Test <FormGroup />", () => {
    describe("Should contain required props", () => {
        const wrapper = mount(<FormGroup {...props} />)
        it("It should contain required props", () => {
            expect(wrapper.prop('type')).toEqual("text")
            expect(wrapper.prop('label')).toEqual("email")
            expect(wrapper.prop('labelName')).toEqual("email")
            expect(wrapper).toMatchSnapshot()
        })
    })
    describe("State should change when click on rear image", () => {
        it("State change when click on rear image", () => {
            const changeState = jest.fn()
            const wrapper = shallow(<FormGroup onClick={changeState} {...props} />)
            const handleClick = jest.spyOn(React, 'useState')
            handleClick.mockImplementation(isShow => [isShow, changeState]);
            const img = wrapper.find("[data-testid='toggle-image']")
            img.simulate('click')
            expect(changeState).toBeTruthy()
        })
        it("Input type change when click on image", () => {
            const changeState = jest.fn()
            const wrapper = shallow(<FormGroup onClick={changeState} {...props} type="password" label="password" labelName="password" />)
            const handleClick = jest.spyOn(React, 'useState')
            handleClick.mockImplementation(isShow => [isShow, changeState]);
            const img = wrapper.find("[data-testid='toggle-image']")
            img.simulate('click')
            const input = wrapper.find('[data-testid="input"]')
            expect(input.props().type).toEqual("text")
        })
    })
})