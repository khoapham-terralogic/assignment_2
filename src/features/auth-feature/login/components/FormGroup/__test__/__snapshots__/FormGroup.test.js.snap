// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <FormGroup /> Should contain required props It should contain required props 1`] = `
<FormGroup
  frontSvg="../Images/frontSvg.svg"
  label="email"
  labelName="email"
  rearSvg="../Images/rearSvg.svg"
  rearSvgShow="../Images/rearSvgShow.svg"
  type="text"
>
  <div
    className="form-group"
  >
    <label
      htmlFor="email"
    >
      email
    </label>
    <div
      className="email-group"
    >
      <img
        alt=""
        className="form-group-frontSvg"
        src="../Images/frontSvg.svg"
      />
      <input
        className="form-control email-group"
        data-testid="input"
        type="text"
      />
      <img
        alt=""
        className="form-group-rearSvg"
        data-testid="toggle-image"
        onClick={[Function]}
        src="../Images/rearSvg.svg"
      />
    </div>
  </div>
</FormGroup>
`;
